File: app/src/main/java/com/shahin/movieapp/MovieApplication.java
package com.shahin.movieapp;

import android.app.Application;

import io.realm.Realm;
import io.realm.RealmConfiguration;

public class MovieApplication extends Application {

    @Override
    public void onCreate() {
        super.onCreate();

        initRealmDB();
    }

    private void initRealmDB() {
        Realm.init(getApplicationContext());
        RealmConfiguration realmConfiguration = new RealmConfiguration.Builder()
                .name("Movies.db")
                .schemaVersion(1)
                .deleteRealmIfMigrationNeeded()
                .build();

        Realm.setDefaultConfiguration(realmConfiguration);
    }

}

File: app/src/main/java/com/shahin/movieapp/app/MovieApplication.kt
package com.shahin.movieapp.app

import android.app.Application
import io.realm.Realm
import io.realm.RealmConfiguration

class MovieApplication : Application() {

    override fun onCreate() {
        super.onCreate()

        initRealmDB()

    }

    private fun initRealmDB() {
        Realm.init(applicationContext)
        val realmConfig = RealmConfiguration.Builder()
            .name("Movies.db")
            .schemaVersion(1)
            .deleteRealmIfMigrationNeeded()
            .build()
        Realm.setDefaultConfiguration(realmConfig)
    }
}